### PR TITLE
Clear up the reference around let

### LINKED ISSUE #29801: reference documentation does not explain "if let" and "while let" clearly
When looking at the reference docs (stable and nightly), the sections on "`if let` expressions" and "`while let` expressions" both refer to "refutable let statements".  I think there are two problems with this description:

1. The phrase "let statements" only appears in the section on "type coercions", and even there its meaning is only hinted at by example.  Maybe this should somehow refer to the section on "declaration statements" (which itself doesn't currently mention `let`)?

2. The phrase "refutable let statements" is nowhere defined in the document.  The section on functions mentions "function arguments are irrefutable patterns", but the notion of refutability doesn't get a good treatment.  There are examples, which do help, but this is not necessarily the same thing as explaining it in prose, and a prose description would provide a place to hyperlink to.

(All this being said, I did clear up my own confusion on this by being able to look at the examples, which are super-helpful.)

### PR DESCRIPTION
**Pull request by steveklabnik**
First, re-word the section on if let/while let to be more clear.

Second, actually call them let statements in the statement section

Fixes #29801

### CODE REVIEW

File: src/doc/reference.md [hunk 1]
```diff
@@ -3211,11 +3212,11 @@ if let ("Ham", b) = dish {
 
 ### `while let` loops
 
-A `while let` loop is semantically identical to a `while` loop but in place of a
-condition expression it expects a refutable let statement. If the value of the
-expression on the right hand side of the let statement matches the pattern, the
-loop body block executes and control returns to the pattern matching statement.
-Otherwise, the while expression completes.
+A `while let` loop is semantically identical to a `while` loop but in place of
+a condition expression it expects `let` statement with a refuatble pattern. If
```
Comment (Manishearth) [inline src/doc/reference.md]: speling

**Comment by Manishearth** [comment]
r=me once the spelling is fixed

File: src/doc/reference.md [hunk 2]
```diff
@@ -3190,10 +3190,11 @@ let message = match maybe_digit {
 
 ### `if let` expressions
 
-An `if let` expression is semantically identical to an `if` expression but in place
-of a condition expression it expects a refutable let statement. If the value of the
-expression on the right hand side of the let statement matches the pattern, the corresponding
-block will execute, otherwise flow proceeds to the first `else` block that follows.
+An `if let` expression is semantically identical to an `if` expression but in
+place of a condition expression it expects a `let` statement with a refuatble
```
Comment (aravind-pg) [inline src/doc/reference.md]: Typo: refuatable --> refutable

**Pull request merged**